core/templates/dev/head/components/button-directives/hint-and-solution-buttons.directive.ts

```diff
@@ -143,8 +134,18 @@ angular.module('oppia').directive('hintAndSolutionButtons', [
               resetLocalHintsArray();
             }
           });
-          resetLocalHintsArray();
-        }
-      ]
+          ctrl.$onInit = function() {
+            ctrl.hintIndexes = [];
+            // Represents the index of the currently viewed hint.
+            ctrl.activeHintIndex = null;
+            ctrl.displayedCard = null;
+            ctrl.solutionModalIsActive = false;
+            ctrl.currentlyOnLatestCard = true;
+            ctrl.isHintConsumed = HintsAndSolutionManagerService.isHintConsumed;
```

Comment:
Same question about functions (this line and the next one). 

(Probably worth doing a little bit of manual testing to check that things work as you expect, both in terms of: declaring the function outside $onInit, and using `= function() {...}` assignment rather than `= ...`.)

Reply:
Replied above.